analysis/webservice/webmodel/CustomEncoder.py

```diff
@@ -0,0 +1,49 @@
+import json
+from datetime import datetime
+from decimal import Decimal
+
+import numpy as np
+
+
+class CustomEncoder(json.JSONEncoder):
+    def default(self, obj):
+        """If input object is an ndarray it will be converted into a dict
+        holding dtype, shape and the data, base64 encoded.
+        """
+        numpy_types = (
+            np.bool_,
+            # np.bytes_, -- python `bytes` class is not json serializable
+            # np.complex64,  -- python `complex` class is not json serializable
+            # np.complex128,  -- python `complex` class is not json serializable
+            # np.complex256,  -- python `complex` class is not json serializable
+            # np.datetime64,  -- python `datetime.datetime` class is not json serializable
```

Comment:
datetimes are quite common in our data - I wonder if we can convert to string?

Reply:
It is a piece of code I moved but I did not change the content. I don't even see where this class is used.

Where do you think we should convert it to a string ? Generally speaking for code simplicity I would keep a datetime object in the code and only convert it from inputs or to outputs but there might  argument against that (size of the object ?).